### PR TITLE
Allow editing shareCompanyPlan

### PR DESCRIPTION
**Pull request by fjvallarino**
@Rise-Vision/apps please review

### CODE REVIEW

File: src/js/controllers/ctr-company-settings-modal.js [hunk 1]
```diff
@@ -43,6 +43,7 @@ angular.module("risevision.common.header")
           $scope.company = company;
           $scope.company.isSeller = company && company.sellerId ? true : false;
           $scope.company.isChargebee = company && company.origin === "Chargebee";
+          $scope.company.ownsPlan = company && ["Active", "Trial"].indexOf(company.planSubscriptionStatus) >= 0;
```
Comment (alex-deaconu) [inline src/js/controllers/ctr-company-settings-modal.js]: @fjvallarino why do we have this extra condition? I don't think this is necessary and we can show the checkbox for any Company regardless of the Plan status. Thanks.

**Comment by alex-deaconu** [comment]
@fjvallarino one comment inline. Don't forget to bump the version before merging. Thanks.

File: src/templates/company-fields.html [hunk 2]
```diff
@@ -121,6 +121,10 @@
     <option value="{{c[1]}}" ng-repeat="c in timezones">{{c[0]}}</option>
   </select>
 </div>
+<div class="form-group" require-role="sa">
```
Comment (alex-deaconu) [inline src/templates/company-fields.html]: @fjvallarino as per discussion, we should probably also update this require-role="ba". Thanks.

File: src/js/controllers/ctr-company-settings-modal.js [hunk 3]
```diff
@@ -156,7 +156,6 @@ angular.module("risevision.common.header")
         //exclude fields from API call
         delete company.sellerId;
         delete company.isTest;
-        delete company.shareCompanyPlan;
```
Comment (alex-deaconu) [inline src/js/controllers/ctr-company-settings-modal.js]: @fjvallarino isn't this still necessary for other roles than RiseStoreAdmin? Thanks.

Comment (fjvallarino) [reply src/js/controllers/ctr-company-settings-modal.js]: @alex-deaconu it actually is; I will restore it. I'm currently testing a couple of things. I'll let you know as soon as the PR is ready for review

**Comment by fjvallarino** [comment]
@alex-deaconu @olegrise this is ready for review again

File: src/js/controllers/ctr-company-settings-modal.js [hunk 4]
```diff
@@ -156,6 +156,7 @@ angular.module("risevision.common.header")
         //exclude fields from API call
         delete company.sellerId;
         delete company.isTest;
+        delete company.shareCompanyPlan;
```
Comment (alex-deaconu) [inline src/js/controllers/ctr-company-settings-modal.js]: @fjvallarino just a note, I had no idea this was a thing. Good pattern to keep in mind for other APIs, especially if we want to fix the Presentation API issue with the template sharing field. Thanks.

**Comment by alex-deaconu** [comment]
@fjvallarino looks good 👍 , thanks for fixing!

**Pull request merged**